runtime/src/accounts.rs

```diff
@@ -547,14 +546,21 @@ impl Accounts {
                     })
                     .map(|(pubkey, account, _slot)| (*pubkey, account.lamports))
                 {
-                    collector.push(data)
+                    let index_of_first_smaller_account =
+                        collector.iter().position(|&(_, balance)| balance <= data.1);
+                    match index_of_first_smaller_account {
+                        Some(i) => {
+                            collector.insert(i, data);
+                        }
+                        None => {
+                            if collector.is_empty() {
```

Comment:
```suggestion
                            if collector.len() <= num {
```

Right?  If the accounts happened to be presorted largest to smallest, this would only return one item.

Reply:
Right, totally. Thx!